### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by marceloneppel**
# Issue
There is no documentation about the events flow inside the charm.

# Solution
Initially add flowcharts for the `peer relation` events and also for the `leader elected` and `pebble ready` events.

# Context
The remaining events (about `db`, `db-admin` and `database` relations still need to be added - in a future PR).

# Testing
N/A

# Release Notes
Add flowcharts for `peer relation` events and also for the `leader elected` and `pebble ready` events.

### CODE REVIEW

File: documentation/reference/charm.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+### PostgreSQL Pebble Ready Hook
+
+```mermaid
+flowchart TD
+  hook_fired([leader-elected Hook]) --> create_pgdata{Create data directory}
+  create_pgdata --> is_leader_or_has_cluster_initialised{Is current\nunit leader or \n has the cluster initialised?}
+  is_leader_or_has_cluster_initialised -- no --> defer>defer]
+  is_leader_or_has_cluster_initialised -- yes --> has_pushed_tls_files{Has successfully \n pushed TLS files?}
+  has_pushed_tls_files -- no --> defer2>defer]
+  has_pushed_tls_files -- yes --> has_services_changed{Has pebble services changed?}
```
Comment (WRFitch) [inline documentation/reference/charm.md:44]:
```suggestion
  has_pushed_tls_files -- yes --> has_services_changed{Have pebble \n services changed?}
```

File: documentation/reference/charm.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  has_services_changed -- no --> has_member_started
+  has_services_changed -- yes --> update_and_restart_service[Update and restart \n the PostgreSQL service]
+  update_and_restart_service --> has_member_started{Have Patroni and PostgreSQL \n started in the current unit?}
```
Comment (WRFitch) [inline documentation/reference/charm.md:47]:
```suggestion
  update_and_restart_service --> has_member_started{Have Patroni and \n PostgreSQL  started in \n the current unit?}
```

File: documentation/reference/charm.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  has_member_started -- no --> defer3>defer]
+  has_member_started -- yes --> is_leader{Is current\nunit leader?}
+  is_leader -- no --> update_config
+  is_leader -- yes --> has_patched_pod_labels{Has successfully \n patched pod labels of \n the new current unit}
```
Comment (WRFitch) [inline documentation/reference/charm.md:51]: This part of the flowchart is a little visually confusing - could you play with it and try to swap the yes/no branches of this decision point?

File: documentation/reference/relations/peers.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  all_units_on_cluster -- no --> are_all_members_ready{Are all cluster \n members ready?}
+  
+  %% This defer should have a return after it to stop the execution.
+  are_all_members_ready -- no --> defer2>defer]
+  defer2 --> is_part_of_cluster
+  
+  are_all_members_ready -- yes --> add_unit_to_cluster[Add unit to cluster]
+  add_unit_to_cluster --> patch_pod_labels[Patch pod labels of the new cluster member]
+  patch_pod_labels --> all_units_on_cluster
+  is_leader -- no --> is_part_of_cluster{Is current unit \n part of the cluster?}
```
Comment (WRFitch) [inline documentation/reference/relations/peers.md:31]: I believe if you move this up to line 20, you might have a slightly cleaner line in your flowchart

File: documentation/reference/relations/peers.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  is_part_of_cluster -- no --> rtn([return])
+  is_part_of_cluster -- yes --> update_config[Update Patroni and \n PostgreSQL config \n]
+  update_config --> restart_postgresql[Restart PostgreSQL \n if TLS is turned on/off]
+  restart_postgresql --> has_member_started{Have Patroni and PostgreSQL \n started in the current unit?}
```
Comment (WRFitch) [inline documentation/reference/relations/peers.md:35]:
```suggestion
  restart_postgresql --> has_member_started{Have Patroni and \n PostgreSQL started in \n the current unit?}
```

**Review by WRFitch — APPROVED** [review]
A bunch of formatting nits, but these look really good! One point I'd mention is that it might help to try and keep these flowcharts relatively simple, but otherwise great :D

**Review by taurus-forever — APPROVED** [review]
Even if it is not perfect, it is better as the current status "nothing" :-D
If possible, please address Will comments and LGTM! Let's merge this to move forward!

File: documentation/reference/relations/peers.md (hunk 4 above)
Comment (marceloneppel) [reply documentation/reference/relations/peers.md]: That's right. Thanks for the tip!

File: documentation/reference/charm.md (hunk 3 above)
Comment (marceloneppel) [reply documentation/reference/charm.md]: Of course. I made a simple change on https://github.com/canonical/postgresql-k8s-operator/pull/78/commits/cb3f332c3197ba4982781fa3a5baa4d2f55132c5. It seems to have fixed the confusion when displaying the flow between the steps.

**Review by WRFitch — APPROVED** [review]

**Review by dragomirp — APPROVED** [review]

**Pull request merged**